articles/css-in-js-solutions/index.md

```diff
@@ -0,0 +1,143 @@
+---
+layout: engineering-education
+status: publish
+published: true
+url: /engineering-education/css-in-js-solutions/
+title: CSS-In-JS For Web Components
+description: This article will cover various techniques a developer can use in defining styles within app components during development than maintaining multiple stylesheets for a single app component.
+author: aransiola-ayodele-leom
+date: 2020-11-11T00:00:00-16:00
+topic: []
+excerpt_separator: <!--more-->
+images:
+
+    - url: /engineering-education/css-in-js-solutions/hero.jpg
+      alt: code view on a laptop
+---
+The modern web is now written in components and not pages. CSS was designed to declare styles in different files (.CSS) for each page of your web application, and this structure makes it challenging to maintain multiple stylesheets for multiple pages.
+<!--more-->
+
+### CSS-In-JS For Web Components
+Web components have changed the landscape of development of web applications for front-end developers. The use of web components provides ample opportunities to reuse and build UI components and custom elements that are critical in the development process.
+
+Understanding web components might seem a bit daunting for novice web developers. However, let’s take a web application that has two prime features― a dropdown menu and a navigation menu appears on all pages.
+
+The traditional way of building web applications would require creating multiple CSS, JavaScript and HTML files that have a dropdown and navigation menu as a feature.
```

Comment:
@CodeLeom, this could be clearer. **See suggested example below from previous review:**

The traditional way of building web applications would require linking JavaScript and CSS files to every page that needed a dropdown or navigation menu and copy/pasting the HTML code to every location you wanted them to appear.

This methodology of building web applications makes it difficult to maintain because in a big application, you could have hundreds of small files or one very large file which is difficult to sort through.